src/main/scala/com/databricks/spark/sql/perf/mllib/feature/VectorAssembler.scala

```diff
@@ -13,35 +13,41 @@ import com.databricks.spark.sql.perf.mllib.{BenchmarkAlgorithm, MLBenchContext,
 /** Object for testing VectorAssembler performance */
 object VectorAssembler extends BenchmarkAlgorithm with TestFromTraining {
 
+  private def getInputCols(numInputCols: Int): Array[String] = {
+    Array.tabulate(numInputCols)(i => s"c${i}")
+  }
+
   override def trainingDataSet(ctx: MLBenchContext): DataFrame = {
     import ctx.params._
-    import ctx.sqlContext.implicits._
 
-    var df = DataGenerator.generateContinuousFeatures(
+    require(numInputCols.get <= numFeatures.get,
+      s"numInputCols (${numInputCols}) cannot be greater than numFeatures (${numFeatures}).")
+
+    val df = DataGenerator.generateContinuousFeatures(
       ctx.sqlContext,
       numExamples,
       ctx.seed(),
       numPartitions,
-      numFeatures * numInputCols
-    )
-    val sliceVec = udf { (v: Vector, from: Int, until: Int) =>
-      Vectors.dense(v.toArray.slice(from, until))
-    }
-    for (i <- (1 to numInputCols.get)) {
-      val colName = s"inputCol${i.toString}"
-      val fromIndex = (i - 1) * numFeatures
-      val untilIndex = i * numFeatures
-      df = df.withColumn(colName, sliceVec(col("features"), lit(fromIndex), lit(untilIndex)))
```

Comment:
The previous implementation slices the vector multiple times, which is slow. The new implementation slices the vector only once.